pages/treeherder.py

```diff
@@ -31,12 +31,14 @@ class TreeherderPage(Base):
     _quick_filter_locator = (By.ID, 'quick-filter')
     _repos_menu_locator = (By.ID, 'repoLabel')
     _result_sets_locator = (By.CSS_SELECTOR, '.result-set:not(.row)')
+    _resultset_visible_locator = (By.CSS_SELECTOR, 'span.hidden')
     _unchecked_repos_links_locator = (By.CSS_SELECTOR, '#repoLabel + .dropdown-menu .dropdown-checkbox:not([checked]) + .dropdown-link')
     _unclassified_failure_count_locator = (By.ID, 'unclassified-failure-count')
     _unclassified_failure_filter_locator = (By.CSS_SELECTOR, '.btn-unclassified-failures')
 
     def wait_for_page_to_load(self):
-        self.wait.until(lambda s: self.unclassified_failure_count > 0)
+        el = self.find_element(*self._resultset_visible_locator)
+        self.wait.until(lambda s: 'hidden' in el.get_attribute('class'))
```

Comment:
The element is located based on the presence of the `hidden` class, so this wait will be immediately true. I'm also concerned that `span.hidden` is very vague and could easily locate other elements that may be added in the future.

Looking at the DOM, I wonder if a better locator might be `#th-global-content span.hidden.ready` (I suspect the `.ready` is important. Alternatively, you could us `#th-global-content span.hidden` and wait for the `class` to include `ready` or the `ng-class` to include `jobsReady`.